app/run.go

```diff
@@ -96,8 +96,10 @@ func (app *App) Run(ctx context.Context) error {
 			backgroundDataSendWg.Wait()
 			if event.EventType == extension.Shutdown {
 				app.logger.Infof("Exiting due to shutdown event with reason %s", event.ShutdownReason)
-				// Flush buffered logs if any
-				app.logsClient.FlushData(ctx, event.RequestID, event.InvokedFunctionArn, app.apmClient.LambdaDataChannel, true)
+				if app.logsClient != nil {
```

Comment:
Alternatively, check `lc` for nil in `FlushData` and return early (slightly reduces number of lines changed).